src/web/mjs/connectors/MangaFire.mjs

```diff
@@ -0,0 +1,163 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class MangaFire extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'mangafire';
+        super.label = 'MangaFire';
+        this.tags = [ 'manga', 'english', 'french', 'japanese', 'portuguese', 'spanish' ];
+        this.url = 'https://mangafire.to';
+        this.path = '/az-list?page=';
+
+        this.queryMangaTitleFromURI = 'main div.top div.info div.name';
+        this.queryMangas = 'div.container div.mangas div.item div.name a';
+        this.idRegex = /manga\/.*\.(\w+)/;
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangaTitleFromURI);
+        const id = uri.pathname.match(this.idRegex)[1];
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getMangas() {
+        const mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangas.length ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const uri = new URL(this.path + page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: element.href.match(this.idRegex)[1],
```

Comment:
The `element.pathname` property might be a better choice as it provides only the least necessary information